module_bom.go

```diff
@@ -75,6 +83,7 @@ func (m ModuleBOM) Generate(workingDir string) ([]packit.BOMEntry, error) {
 		return nil, fmt.Errorf("failed to decode bom.json: %w", err)
 	}
 
+	var packageLock packageLock
```

Comment:
We probably don't want a type and a variable with the same identifier. The first declaration works, but any other instantiation of the `packageLock` type within that same scope will fail. Here is a toy example: https://play.golang.org/p/E1iYQViujJD